state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2838,15 +2824,43 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		}
 	}
 
+	fn refund_invalid_ccm(
+		ccm: CrossChainMessage<T::TargetChain, BlockNumberFor<T>>,
+		error: ExecutexSwapAndCallError,
+	) {
+		Self::deposit_event(Event::<T, I>::CcmEgressInvalid { egress_id: ccm.egress_id, error });
+		if let Ok(decoded_data) = T::CcmValidityChecker::decode_unchecked(
+			ccm.ccm_additional_data.clone(),
+			T::TargetChain::get(),
+		) {
+			if let Some(fallback_address) = decoded_data.refund_address::<T::TargetChain>() {
+				match Self::schedule_egress(
+					ccm.asset,
+					ccm.amount,
+					fallback_address.clone(),
+					None,
+					None
+				) {
+					Ok(egress_details) => Self::deposit_event(Event::<T, I>::InvalidCcmRefunded {
```

Comment:
I'm not sure this will be enough information in the event for frontends to link things together easily. May need to include the egress_id too. CC: @acdibble

Reply:
I've added the egress_id in anyway

Reply:
Got it. I've added the new egress ID as well - so now both IDs are there in the event.